src/main.rs

```diff
@@ -214,19 +206,24 @@ fn queue_render(
         .par_chunks_exact_mut(4)
         .zip(INDEX)
         .for_each(|(pixel, i)| {
+            // SAFETY: Pixel size will always be 4, RGBA
+            unsafe {
+                std::intrinsics::assume(pixel.len() == 4);
+            }
+
             // (x, y) of pixel on screen
-            let (x, y): (i32, i32) = (((i % DIMS.0) as i32), ((i / DIMS.0) as i32));
+            let (x, y): (u32, u32) = ((i % DIMS.0), (i / DIMS.1));
```

Comment:
This was correct originally.
```suggestion
            let (x, y): (u32, u32) = ((i % DIMS.0), (i / DIMS.0));
```